pkg/build/config/version.go

```diff
@@ -0,0 +1,79 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"os"
+)
+
+type Metadata struct {
+	GrafanaVersion string `json:"version,omitempty"`
+	ReleaseMode    string `json:"releaseMode,omitempty"`
+	GrabplVersion  string `json:"grabplVersion,omitempty"`
+}
+
+type PluginSignature struct {
+	Sign      bool `json:"sign,omitempty"`
+	AdminSign bool `json:"adminSign,omitempty"`
+}
+
+type Docker struct {
+	ShouldSave    bool           `json:"shouldSave,omitempty"`
+	Architectures []Architecture `json:"archs,omitempty"`
+}
+
+// Version represents the "version.json" that defines all of the different variables used to build Grafana
+type Version struct {
+	Variants                  []Variant       `json:"variants,omitempty"`
+	PluginSignature           PluginSignature `json:"pluginSignature,omitempty"`
+	Docker                    Docker          `json:"docker,omitempty"`
+	PackagesBucket            string          `json:"packagesBucket,omitempty"`
+	PackagesBucketEnterprise2 string          `json:"packagesBucketEnterprise2,omitempty"`
+	CDNAssetsBucket           string          `json:"CDNAssetsBucket,omitempty"`
+	CDNAssetsDir              string          `json:"CDNAssetsDir,omitempty"`
+	StorybookBucket           string          `json:"storybookBucket,omitempty"`
+	StorybookSrcDir           string          `json:"storybookSrcDir,omitempty"`
+}
+
+// Versions is a map of versions. Each key of the Versions map is an event that uses the the config as the value for that key.
+// For example, the 'pull_request' key will have data in it that might cause Grafana to be built differently in a pull request,
+// than the way it will be built in 'main'
+type VersionMap map[string]Version
+
+// GetMetadata attempts to read the JSON file located at 'path' and decode it as a Metadata{} type.
+func GetMetadata(path string) (*Metadata, error) {
+	// Ignore gosec G304 as this function is only used in the build process.
+	//nolint:gosec
+	file, err := os.Open(path)
```

Comment:
Just for my information - why is `os.Open` better than what exists already in the build-pipeline repo?

Reply:
It does all of the same things but with less code.

The 'GetMetadata' function in grabpl:

* Checks if the directory exists
    * If it doesn't exist, then returns an error
    * If it does exist, then it adds "version.json" to the path.
    * If that file doesn't exist, it returns an error.
    * It reads all of the versionJSON data into a byte array, then unmarshals it.

`os.Open` will:
* return an error if the directory or file don't exist
* returns an io.Reader, which can be used in `json.NewDecoder(r).Decode(&Metadata)`, which saves us from having to do both an `io.ReadAll` and json.Unmarshal.

Reply:
I see. Thought that the question here is whether the json is known and stored, or comes from a request. 
In the former case I use `json.Unmarshal`, in the latter I use `json.NewDecoder().Decode()`.

Reply:
No need to change it though, I was just curious.

Reply:
Nah, funcitonally they do the same thing, only json.Unmarshal expects a byte array whereas a decoder expects a reader to get its bytes. So with Unmarshal you need to do something like io.ReadAll to get all the bytes first.

You'd might use `json.NewDecoder` with a request because `req.Body` also satisfies `io.Reader`, just like an `os.File` does.